.github/workflows/api_diff_report.yml

```diff
@@ -0,0 +1,90 @@
+name: API Diff Report
+
+on: [pull_request]
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name }}-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+env:
+  STAGE_PROGRESS: progress
+  STAGE_END: end
+  PR_API_OUTPUT: ci_outputs/pr_branch_api
+  BASE_API_OUTPUT: ci_outputs/base_branch_api
+  DIFF_REPORT_OUTPUT: ci_outputs/diff_report
+
+jobs:
+  diff_report:
+    runs-on: macos-latest
+
+    steps:
+      - name: Checkout PR branch
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 2
+
+      - name: Copy diff report tools
+        run: cp -a scripts/api_diff_report/. ~/api_diff_report
```

Comment:
what is the purpose of copying the diff report tools?

Reply:
We are checkout 2 version of code. The tools in base branch may get outdated and causing issue in the workflow.

Reply:
As asked in https://github.com/firebase/firebase-ios-sdk/pull/11236#discussion_r1186571006, if we check out base code at a different location, and as a result the diff report tool no longer gets overwritten, do we still need the copy here?